src/target.rs

```diff
@@ -2,129 +2,52 @@
 //!
 //! See the `target-lexicon` crate for a more principled approach to this.
 
-use std::{borrow::Cow, env, str::FromStr};
+use std::str::FromStr;
 
 use crate::{Error, ErrorKind};
 
 mod apple;
 mod generated;
 mod llvm;
+mod parser;
+
+pub(crate) use parser::TargetInfoParser;
 
 /// Information specific to a `rustc` target.
 ///
 /// See <https://doc.rust-lang.org/cargo/appendix/glossary.html#target>.
 #[derive(Debug, PartialEq, Clone)]
-pub(crate) struct TargetInfo {
+pub(crate) struct TargetInfo<'a> {
```

Comment:
cc @madsmtm This is the optimization I was proposing, can you take a look and confirm this looks good please?